src/mon/NVMeofGwSerialize.h

```diff
@@ -254,20 +255,38 @@ inline void decode(ana_state_t& st, ceph::buffer::list::const_iterator &bl) {
   DECODE_FINISH(bl);
 }
 
-inline void encode(const GwSubsystems& subsystems,  ceph::bufferlist &bl) {
-  ENCODE_START(1, 1, bl);
+inline void encode(
+  const GwSubsystems& subsystems,  ceph::bufferlist &bl, uint64_t features) {
+  uint8_t version = 1;
+  if (HAVE_FEATURE(features, NVMEOFHA)) {
+    version = 2;
+  }
+  ENCODE_START(version, version, bl);
```

Comment:
I understand that if you encode at version 2, it is only decodable by version 2, but I recommend creating a min_compat variable now so that in the future you don't inadvertently require version 3 to decode version 3.